wp-content/themes/mstoday/functions.php

```diff
@@ -156,3 +156,111 @@ function mstoday_add_wide_image_support() {
   add_theme_support( 'align-wide' );
 }
 add_action( 'after_setup_theme', 'mstoday_add_wide_image_support' );
+
+/**
+ * A shim to make captions in articles show as actual captions instead of as body content.
+ * 
+ * Waiting for this to be fixed in the actual ANP plugin. 
+ * https://github.com/INN/umbrella-mstoday/issues/57#issuecomment-580844062
+ * 
+ * The general idea here is to find all images attached to the post, find their
+ * captions and media credits, then loop through $json and if any body component matches
+ * an image caption/credit, change that component to be `role = caption` instead of `role = body`.
+ * 
+ * @param arr $json An array of the post content in JSON format
+ * @param int $post_id The post id
+ * 
+ * @return arr @json The modified post content JSON
+ */
+function mstoday_apple_news_article_update_captions_shim( $json, $post_id ){
+
+	// grab all attachments tied to this post
+	$attachments = get_attached_media( 'image', $post->ID );
+
+	// init empty arr
+	$captions_and_credits = array();
+
+	// loop through all returned attachments and add their captions and 
+	// credits to our $captions_and_credits arr
+	foreach( $attachments as $attachment ){
+
+		$attachment_caption = wp_get_attachment_caption( $attachment->ID );
+
+		if ( function_exists( 'navis_get_media_credit' ) ) {
+			$attachment_credit = navis_get_media_credit( $attachment->ID );
+		}
+
+		if( !empty( $attachment_caption ) ) {
+			array_push( $captions_and_credits, preg_replace("/[^a-z0-9.]+/i", "", $attachment_caption ) );
+		}
+
+		if( isset( $attachment_credit ) && !empty( $attachment_credit ) ) {
+			if( !empty( $attachment_credit->credit ) ) {
+				array_push( $captions_and_credits, preg_replace("/[^a-z0-9.]+/i", "", $attachment_credit->credit ) );
+			}
+
+			if( !empty( $attachment_credit->org ) ) {
+				array_push( $captions_and_credits, preg_replace("/[^a-z0-9.]+/i", "", $attachment_credit->org ) );
+			}
+		}
+
+	}
+
+	// woohoo set an index
+	$index = -1;
+	$dropcap_set = false;
+
+	// this is where the fun begins
+	// loop through all $json body components and if one matches a caption/credit, update its role
+	foreach( $json['components'] as $component ) {
+
+		$index++;
+
+		// we have to make it only alphanumeric because this plugin is annoying and adds way too much whitespace
+		// that doesn't want to be stripped out with trim, str_replace, etc.
+		$component_text = preg_replace("/[^a-z0-9.]+/i", "", $component['text']);
+		$component_text = trim( $component_text, 'p' );
+	
+		// if the component text actually matches a credit or caption, update it
+		if( in_array( $component_text, $captions_and_credits ) ) {
```

Comment:
This will fail if the caption in the `post_content` does not match the caption set upon the attachment, which is possible because people can edit the `[caption]` shortcode contents or other parts of the `post_content`.

But I don't have a better answer to this, because we don't have `'role'` as `'caption'` set on this component.